### ISSUE TITLE
Feature Request: Resident Key Management

### ISSUE THREAD

**Issue by MichaelGrafnetter** [conorpp/solo-desktop#18]
It would be cool if it were possible to view/delete resident credentials.

**Comment by minghuadev** [comment]
An [earlier issue 156 on solo](https://github.com/solokeys/solo/issues/156) seems to agree on not to make that available. See this by @nickray on Mar 28 2019: 
> IMHO, the resident key interface isn't so well thought out by FIDO: You can't delete a specific key (nor list all resident keys on the device, which makes some sense from a security perspective if you lose the key), so if you run out of space, you have to reset the Solo key and delete all of the resident keys. (One exception, you can replace the key for a given (relying party, user ID) pair.)

And this by  @0x0ece on Mar 28 2019:
> (I agree list, single delete shouldn't be available for security reasons -- though it's still possible to enumerate providing rpid.)

Does anybody have an example of what other products are doing and how it is used?

**Comment by MichaelGrafnetter** [comment]
Keys from Yubico and Feitian support this feature, and so does libfido2. 

Listing RPs using libfido2 on Windows:
![image](https://user-images.githubusercontent.com/8986376/90972238-a0065700-e517-11ea-915e-f91bcaf2510b.png)

Listing RKs for a RP (webauthn.io), deleting one of them, and checking the result:
![image](https://user-images.githubusercontent.com/8986376/90972272-dfcd3e80-e517-11ea-8c88-ebff12e541d8.png)

RK management makes a lot of sense to me, especially in the following scenarios:
- I would like my secondary Solo Key to be registered with all the accounts with which my primary one is registered. Have I missed a RP or an account?
- I would like to replace my old key with a new one that has a better form factor. With what services have I registered my original key?
- My key contains many test accounts, but also a couple of production ones. I therefore need to perform selective deletion.

**Comment by MichaelGrafnetter** [comment]
BTW, the [authenticatorCredentialManagement feature](https://fidoalliance.org/specs/fido2/fido-client-to-authenticator-protocol-v2.1-rd-20191217.html#authenticatorCredentialManagement) is part of the CTAP standard. Specifically, RC deletion is implemented in libfido2 through the [fido_credman_del_dev_rk function](https://github.com/Yubico/libfido2/blob/master/src/fido/credman.h).

And here is [Yubico's accouncement of the Credential Management feature support](https://support.yubico.com/support/solutions/articles/15000027138-yubikey-5-2-3-enhancements-to-fido-2-support).

**Comment by nickray** [comment]
Yes, that's an old comment of mine. Meanwhile, CTAP v2.1 specifies RK management.

It is available if you update firmware to v4 (https://github.com/solokeys/solo/releases/tag/4.0.0), which is not yet the default for sold keys, as we'd have to re-certify with the FIDO Alliance.

**Comment by MichaelGrafnetter** [comment]
Thx, @nickray . I have just tested the credential management feature through libfido2 with my SoloKey with v4 firmware and it really works seamlessly.
I thus wonder if it could please be exposed through solo-desktop, to make the RK management even easier.

**Comment by minghuadev** [comment]
Thanks for the information. I see the commits are tracked by [https://github.com/solokeys/solo/issues/314](https://github.com/solokeys/solo/issues/314) .
